frontend/src/helpers/dataOverlay.js

```diff
@@ -0,0 +1,97 @@
+const DEFAULT_GENE_COLOR = '#feb';
+const DEFAULT_METABOLITE_COLOR = '#9df';
+
+const DATA_TYPES_COMPONENTS = {
+  transcriptomics: {
+    componentType: 'gene',
+    className: 'enz',
+    defaultColor: DEFAULT_GENE_COLOR,
+  },
+  metabolomics: {
+    componentType: 'metabolite',
+    className: 'met',
+    defaultColor: DEFAULT_METABOLITE_COLOR,
+  },
+};
+
+/*
+ * @param { Blob } file
+ *
+ * @example
+ * try {
+ *   const data = await parseCustomFile(file);
+ * } catch (e) {
+ *   console.error(e)
+ * }
+ */
+const parseFile = async (file) => {
+  let tissues = [];
+  const levels = { 'n/a': 'n/a' };
+
+  const text = await file.text();
+
+  const lines = text.split(/\r?\n/);
+  let indexLine = 1;
+  // fetch Tissue
+  if (lines[0].split('\t').length !== 1) {
+    const arrLine = lines[0].split('\t');
+    const v = Number(arrLine[1]);
+    if (Number.isNaN(v)) {
+      tissues = lines[0].split('\t');
+      tissues.shift();
+      lines.shift();
+    } else {
+      tissues = [];
+      for (let i = 1; i < arrLine.length; i += 1) {
+        tissues.push(`serie${i}`);
+      }
+    }
+  } else {
+    throw new Error('Invalid TSV format, expect at least two columns.');
+  }
+
+  // parse lines
+  // make tissues key;
+  for (let i = 0; i < tissues.length; i += 1) {
+    const tissue = tissues[i];
+    if (tissue in levels) {
+      throw new Error(`Error: duplicated column '${tissue}'.`);
+    }
```

Comment:
This is a check/validation, so it might make sense to move it to the validation component.

Reply:
Great that you noticed these. For reference, these are mostly untouched/slightly adjusted based on [RNAexpression.vue](https://github.com/MetabolicAtlas/MetabolicAtlas/blob/develop/frontend/src/components/explorer/mapViewer/RNAexpression.vue), which has been removed in this PR. Your comments are good to leave up until they can be resolved.